### DISCUSSION TITLE
Long representation in the schema is not coming up as java.lang.Long in the POJOs created and instead coming as String; Is there a specific version with the fix for this?

### DISCUSSION THREAD

**Discussion by siva54** [kobylynskyi/graphql-java-codegen#930] [Q&A] [answered]
Long representation in the schema is not coming up as java.lang.Long in the POJOs created and instead of coming as String; 
Is there a specific version with the fix for this?

I've been doing manual override in the pom.xml through "customTypesMapping" configuration to overcome this and would to avoid going that way.

**Comment by kobylynskyi** [comment]
Please share your schema

**Reply by siva54** [reply]
We have something like,
```
type Query {
 updatedAt: Long!
}
```
This represents the millis since the epoch. This is getting generated as String in the POJO instead of a long.

**Reply by siva54** [reply]
Hi @kobylynskyi . As of now, I'm manually overriding String to Long using a customTypesMapping, After some point in time, This is getting messy with a lot of custom mappings. Is there a way to get Long instead of a String?

**Comment by kobylynskyi** [comment, marked as answer]
@siva54 this is not supported and will not be until GraphQL specification allows such scalar:
https://graphql.org/learn/schema/#scalar-types
As you can see o the spec above:
```
 GraphQL comes with a set of default scalar types out of the box:

    Int: A signed 32‐bit integer.
    Float: A signed double-precision floating-point value.
    String: A UTF‐8 character sequence.
    Boolean: true or false.
    ID: The ID scalar type represents a unique identifier, often used to refetch an object or as the key for a cache. The ID type is serialized in the same way as a String; however, defining it as an ID signifies that it is not intended to be human‐readable.
```
There's no such out-of-the-box scalar as `Long`.
If it is your own scalar (`scalar Long`) then you need to add it to `customTypesMapping` config.

**Reply by siva54** [reply]
Thank you @kobylynskyi . This is nice.
